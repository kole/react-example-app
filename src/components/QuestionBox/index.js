import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Question from '~/components/Question';

const QuestionBox = styled.div`
    align-items: center;
    border: 1px solid #000;
    display: flex;
    font-size: 2em;
    line-height: 1.8em;
    padding: 2em;
`;

const QuestionComponent = (props) => {
    const { questionText } = props;
    return (
        <QuestionBox>
            <Question questionText={questionText} />
        </QuestionBox>
    );
}

QuestionComponent.propTypes = {
    questionText: PropTypes.string.isRequired
}

export default QuestionComponent;
