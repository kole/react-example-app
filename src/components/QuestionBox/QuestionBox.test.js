/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import QuestionBox from './index.js';

describe('QuestionBox', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<QuestionBox questionText="Test question box" />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
