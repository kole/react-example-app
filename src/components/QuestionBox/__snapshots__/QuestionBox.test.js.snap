// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionBox Snapshot 1`] = `
<div
  className="sc-bwzfXH eVoLQX"
>
  <div
    className="sc-bdVaJa UhWAf"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test question box",
      }
    }
  />
</div>
`;
